invisible_cities/cities/icaro.py

```diff
@@ -5,31 +5,51 @@
 
 
 """
-from operator import attrgetter
 
+import numpy as np
+import pandas as pd
 import tables as tb
 
-from .. reco         import tbl_functions as tbl
-from .. io.krmaps_io import write_krmap
-from .. io.krmaps_io import write_krevol
-from .. io.krmaps_io import write_mapinfo
+from .. reco                import tbl_functions as tbl
+from .. reco.corrections    import ASectorMap
+from .. reco.corrections    import maps_coefficient_getter
+from .. reco.corrections    import correct_geometry_
+from .. reco.corrections    import norm_strategy
+from .. reco.corrections    import get_normalization_factor
+from .. reco.corrections    import apply_all_correction
+from .. io.krmaps_io        import write_krmap
+from .. io.krmaps_io        import write_krevol
+from .. io.krmaps_io        import write_mapinfo
+from .. core                import fit_functions as fitf
+from .. core.core_functions import in_range
+from .. core.core_functions import shift_to_bin_centers
 
 from .. dataflow          import dataflow as fl
 from .. dataflow.dataflow import push
 from .. dataflow.dataflow import pipe
 
+from .. types.ic_types    import Measurement
+from .. types.ic_types    import MasksContainer
+
 from .  components import city
 from .  components import print_numberofevents
 from .  components import dst_from_files
 from .  components import quality_check
 from .  components import kr_selection
 from .  components import map_builder
 from .  components import add_krevol
+from .  components import print_every
+
+from typing import Tuple
+from typing import List
+from pandas import DataFrame
 
 
 @city
-def icaro(files_in, file_out, compression, detector_db, run_number,
-          bootstrap, quality_ranges, band_sel_params, map_params, krevol_params):
+def icaro(files_in, file_out, compression, event_range, print_mod,
+          detector_db, run_number, bootstrap, quality_ranges,
+          band_sel_params, map_params,
+          r_fid, nStimeprofile, x_range, y_range):
```

Comment:
Just to check, shouldn't we group the variables in dicts in the config file? I think this is easier to read but not strongly opinionated.

Reply:
Not sure, I just took a look at other cities and I saw that it was done this way. You mean grouping `r_fid` and `nStimeprofile` in a `krevol_params` dictionary and then call the function like this:
```python
add_krevol(**krevol_params)
```
?

Reply:
Yes, something of that sort, but again, it's something we can discuss in the meeting, not a big deal for me.

Reply:
I would only use dictionaries as arguments whenever there are large groups of parameters that impact a specific part of the flow. For instance, if your arguments affect _only_ the time evolution and they are quite a few, I would group them into a dictionary, otherwise, I would leave them as individual arguments.

There is no clear cut though, so waiting for the city to be more advanced will give us a better perspective.